quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java

```diff
@@ -37,11 +44,56 @@
 
     public static final String NAME = "start";
 
-    @CommandLine.Option(names = {AUTO_BUILD_OPTION_SHORT, AUTO_BUILD_OPTION_LONG },
+    @CommandLine.Option(names = {AUTO_BUILD_OPTION_SHORT, AUTO_BUILD_OPTION_LONG},
             description = "Automatically detects whether the server configuration changed and a new server image must be built" +
                     " prior to starting the server. This option provides an alternative to manually running the '" + Build.NAME + "'" +
                     " prior to starting the server. Use this configuration carefully in production as it might impact the startup time.",
             paramLabel = NO_PARAM_LABEL,
             order = 1)
     Boolean autoConfig;
+
+    @Override
+    protected void doBeforeRun() {
+        checkIfProfileIsNotDev();
+    }
+
+    /**
+     * Checks if the profile provided by either the current argument, the system environment or the persisted properties is dev.
+     * Fails with an error when dev profile is used for the start command, or continues with the found profile if its not the dev profile.
+     */
+    private void checkIfProfileIsNotDev() {
+        List<String> currentCliArgs = spec.commandLine().getParseResult().expandedArgs();
+        String profileLongArg = "--profile";
+        String profileShortArg = "-pf";
+
+        String currentArgsProfile = currentCliArgs.stream().filter(arg -> arg.contains(profileLongArg)).findFirst()
```

Comment:
We also need to support options using a space as the key/value separator: `./kc.sh [-pf][--profile] dev`.

Do you really need to query the arg list to query the profile? It should be enough to obtain it from `Environment.getProfile` because the profile should be set prior to execute the start command.

It should help to make it simpler.

Reply:
For point 1: yes, that's right... and adds new challenges as these are 2 different args in cliArgs when set via whitespace. 

For point 2: Yes, I need to. Environment.getProfile only considers the profile from a System variable or env var. To be more precise: if we want to make `./kc.sh start --profile/-pf <profile>` a possibility, we  need to check for a profile which is just added via the current invocation, which we do not, yet. If we don't want this, we should remove the profile command from the start command. Previous behaviour was always defaulting to `prod` even when doing e.g. `kc.sh start --profile=test` bc the actual profile was not set beforehand and this is quarkus standard behaviour.